### PR TITLE
426 Add my groups overview

### LINKED ISSUE #426: My groups
Provide a "my groups" user route to list all the groups the user is a member.

### PR DESCRIPTION
**Pull request by MPParsley**
Fixes #426.

Here's what it would look like.

groups admin theme
groups frontend theme

### CODE REVIEW

File: og.routing.yml [hunk 1]
```diff
@@ -43,3 +43,17 @@ og.remove_multiple_roles_confirm:
...
+og.user_memberships:
+  path: '/user/{user}/groups'
+  defaults:
+    _entity_list: 'og_membership'
+    _title: 'My groups'
+  requirements:
+    user: '\d+'
+    _entity_access: 'user.view'
+  options:
+    _admin_route: 'TRUE'
```
Comment (MPParsley) [inline og.routing.yml]:
```suggestion
```

**Review by pj-claeys — APPROVED** [review]

**Comment by damienmckenna** [comment]
Nice work!

**Comment by damienmckenna** [comment]
Should this be deprecated in favor of #231? They both provide a handler for the list_builder item on the og_membership entity.

**Comment by pfrenssen** [comment]
@MPParsley can you clarify this? Shall we go for the list handler in this PR or in #231?

**Comment by MPParsley** [comment]
@pfrenssen, we should pursue this one.

The list builder in #231 is just the one from core: ` *     "list_builder" = "Drupal\Core\Entity\EntityListBuilder",`

File: src/OgMembershipListBuilder.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Drupal\og;
```
Comment (MPParsley) [inline src/OgMembershipListBuilder.php:5]: Add strict types
```suggestion
declare(strict_types = 1);

namespace Drupal\og;
```

**Comment by claudiu-cristea** [comment]
Since https://www.drupal.org/project/og/issues/3467694 was merged into the `8.x-1.x` branch, the development has been moved to https://www.drupal.org/project/og. If this change is still actual, please open an issue at https://www.drupal.org/project/issues/og and provide there a _merge request_ similar with this pull request.

Thank you for contributing to Organic groups module.

**Pull request closed without merging**